yarn-project/acir-simulator/src/client/execution_result.test.ts

```diff
@@ -0,0 +1,211 @@
+    fnB.nestedExecutions.push(fnC);
+    executionResult.nestedExecutions.push(fnB);
+    const encryptedLogs = collectEncryptedLogs(executionResult);
+    expect(encryptedLogs).toEqual([
+      FunctionL2Logs.empty(),
+      new FunctionL2Logs([Buffer.from('Log 1')]),
+      FunctionL2Logs.empty(),
+    ]);
+  });
+
+  it('collect encrypted logs with no logs in any nested calls', () => {
+    /*
+    Create the following executionResult object:
+    fnA ()
+      |----------> fnB () -> fnC ()
+      |----------> fnD () -> fnE ()
+    Circuits and ACVM process in a DFS + stack like format: [fnA, fnD, fnE, fnB, fnC]
+    */
+    const executionResult: ExecutionResult = emptyExecutionResult();
+    const fnB = emptyExecutionResult();
+    const fnC = emptyExecutionResult();
+    const fnD = emptyExecutionResult();
+    const fnE = emptyExecutionResult();
+
+    fnB.nestedExecutions.push(fnC);
+    fnD.nestedExecutions.push(fnE);
+
+    executionResult.nestedExecutions.push(fnB, fnD);
+
+    const encryptedLogs = collectEncryptedLogs(executionResult);
+    expect(encryptedLogs).toEqual([
+      FunctionL2Logs.empty(),
+      FunctionL2Logs.empty(),
+      FunctionL2Logs.empty(),
+      FunctionL2Logs.empty(),
+      FunctionL2Logs.empty(),
+    ]);
+  });
+});
+
+describe('collect unencrypted logs', () => {
+  // collection of unencrypted logs work similar to encrypted logs, so lets write other kinds of test cases:
+
+  function emptyExecutionResultWithUnencryptedLogs(unencryptedLogs = FunctionL2Logs.empty()): ExecutionResult {
+    const executionResult = emptyExecutionResult();
+    executionResult.unencryptedLogs = unencryptedLogs;
+    return executionResult;
+  }
+
+  it('collect unencrypted logs even when no logs and no recursion', () => {
+    // fnA()
+    const executionResult: ExecutionResult = emptyExecutionResult();
+    const unencryptedLogs = collectUnencryptedLogs(executionResult);
+    expect(unencryptedLogs).toEqual([FunctionL2Logs.empty()]);
+  });
+
+  it('collect unencrypted logs with no logs in some nested calls', () => {
+    /*
+    Create the following executionResult object: 
+    fnA () 
+        |----------> fnB () -> fnC (log1, log2, log3)
+    Circuits and ACVM process in a DFS + stack like format: [fnA, fnC, fnB]
+    */
+    const executionResult: ExecutionResult = emptyExecutionResult();
+    const fnB = emptyExecutionResult();
+    const fnC = emptyExecutionResultWithUnencryptedLogs(
+      new FunctionL2Logs([Buffer.from('Log 1'), Buffer.from('Log 2'), Buffer.from('Log 3')]),
+    );
+
+    executionResult.nestedExecutions.push(fnB, fnC);
+
+    const unencryptedLogs = collectUnencryptedLogs(executionResult);
+    expect(unencryptedLogs).toEqual([
+      FunctionL2Logs.empty(),
+      new FunctionL2Logs([Buffer.from('Log 1'), Buffer.from('Log 2'), Buffer.from('Log 3')]),
+      FunctionL2Logs.empty(),
+    ]);
+  });
+
+  it('collect unencrypted logs with multiple logs each function call', () => {
+    /*
```

Comment:
I would rephrase the name of this test as fnA and fnB do not emit logs. Maybe sthg like:
'collect unencrypted logs with multiple logs in each function call leaves'